src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1896,5 +1906,27 @@ internal void SetMaxLanguageVersion(ProjectId projectId, string maxLanguageVersi
                 _projectToMaxSupportedLangVersionMap[projectId] = maxLanguageVersion;
             }
         }
+
+#nullable enable
+
+        internal void RefreshProjectExistsUIContextForLanguage(string language)
+        {
+            lock (_gate)
+            {
+                var uiContext =
+                    _languageToProjectExistsUIContext.GetOrAdd(
+                        language,
+                        l => Services.GetLanguageServices(l).GetService<IProjectExistsUIContextProviderLanguageService>()?.GetUIContext());
+
+                if (uiContext != null)
```

Comment:
Perhaps add an assert or throw exception if unicontext is null for C# and VB? It would be good to have some error so we could diagnose cases where the uicontext didn't fire...

Reply:
The tests would fail then.

Reply:
So `uiContext` can be legitimately null when language is CSharp or VisualBasic?

Reply:
No, but we usually don't add language-specific asserts like that.